notificationworkerlambda/src/test/scala/com/gu/notifications/worker/HarvesterRequestHandlerSpec.scala

```diff
@@ -0,0 +1,133 @@
+package com.gu.notifications.worker
+
+import java.util.UUID
+
+import _root_.models.Importance._
+import _root_.models.Link._
+import _root_.models.TopicTypes._
+import _root_.models._
+import cats.effect.IO
+import com.amazonaws.services.lambda.runtime.events.SQSEvent
+import com.amazonaws.services.lambda.runtime.events.SQSEvent.SQSMessage
+import com.gu.notifications.worker.models.SendingResults
+import com.gu.notifications.worker.tokens.{ChunkedTokens, SqsDeliveryService, TokenService}
+import com.gu.notifications.worker.utils.Cloudwatch
+import fs2.{Sink, Stream}
+import org.specs2.matcher.Matchers
+import org.specs2.mutable.Specification
+import org.specs2.specification.Scope
+import play.api.libs.json.Json
+
+import scala.collection.JavaConverters._
+
+class HarvesterRequestHandlerSpec extends Specification with Matchers {
+
+  "the WorkerRequestHandler" should {
+    "Send one Android breaking news notification" in new WRHSScope {
+      workerRequestHandler.handleHarvesting(sqsEventShardNotification(breakingNewsNotification, Android), null)
+      tokenStreamCount shouldEqual 1
+      firebaseSqsDeliveriesCount shouldEqual 1
+      apnsSqsDeliveriesCount shouldEqual 0
+
+    }
+    "Queue one Android content notification" in new WRHSScope {
+      workerRequestHandler.handleHarvesting(sqsEventShardNotification(contentNotification, Android), null)
+      tokenStreamCount shouldEqual 1
+      firebaseSqsDeliveriesCount shouldEqual 1
+      apnsSqsDeliveriesCount shouldEqual 0
+    }
+    "Send one iOS breaking news notification" in new WRHSScope {
+      workerRequestHandler.handleHarvesting(sqsEventShardNotification(breakingNewsNotification, iOS), null)
+      tokenStreamCount shouldEqual 1
+      firebaseSqsDeliveriesCount shouldEqual 0
+      apnsSqsDeliveriesCount shouldEqual 1
+
+    }
+    "Queue one iOS content notification" in new WRHSScope {
```

Comment:
Maybe add a test to check the chunking? In it's current form, and with a mixed input (iOS and Android tokens) the chunks will be smaller than 1K, when we expect it to be exactly 1k.